packages/backend/src/apps/custom-api/actions/http-request/index.ts

```diff
@@ -58,34 +55,40 @@ const action: IRawAction = {
     const data = $.step.parameters.data as string
     const url = $.step.parameters.url as string
 
-    // Prohibit calling ourselves to prevent self-DoS.
-    if (new URL(url).hostname.toLowerCase().endsWith('plumber.gov.sg')) {
-      throw new StepError(
-        RECURSIVE_WEBHOOK_ERROR_NAME,
-        RECURSIVE_WEBHOOK_ERROR_SOLUTION,
-        $.step.position,
-        $.app.name,
-      )
-    }
-
-    if (!(await isUrlAllowed(url))) {
-      throw new Error('The URL you are trying to call is not allowed.')
-    }
-
     try {
-      const response = await $.http.request({
+      let response = await $.http.request({
         url,
         method,
         data,
-        // there's an internal maxRedirect limit for chrome/safari,
-        // so we only block redirect back to plumber urls
-        beforeRedirect: (options) => {
-          if (options.hostname.toLowerCase().endsWith('plumber.gov.sg')) {
-            throw new Error(RECURSIVE_WEBHOOK_ERROR_NAME)
-          }
-        },
+        maxRedirects: 0,
+        //  overwriting this to allow redirects to resolve
+        validateStatus: (status) =>
+          (status >= 200 && status < 300) ||
+          REDIRECT_STATUS_CODES.includes(status),
       })
 
+      if (!response) {
+        throw new Error('No response returned')
+      }
+
+      /**
+       * We handle redirects here manually so we could apply the same request interceptors
+       * i.e. checking if url is recursive or resolves to internal ip
+       * this means that we allow for only one hop of redirect
+       */
```

Comment:
I'm sure you've thought of this, but would it be possible/easier to ban redirects outright?

i.e. Onus is on user to supply a valid URL

Reply:
[for such cases](https://developers.google.com/apps-script/guides/content#redirects)